pkgs/development/python-modules/execnb/default.nix

```diff
@@ -0,0 +1,33 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, fastcore
+, traitlets
+, ipython
+, pythonOlder
+}:
+
+buildPythonPackage rec {
+  pname = "execnb";
+  version = "0.1.4";
+  disabled = pythonOlder "3.6";
```

Comment:
Please add `format = ...` (generally "pyproject", which might require adding `setuptools` to `nativeBuildInputs` or "setuptools") to these package.

Reply:
I did a search in nixpkgs. I have seen format that is set to pyproject and setuptools. what's the difference?

Reply:
`pyproject` is for projects that use `pyproject.toml`, while `setuptools` is for other projects with `setup.py` that can be built with setuptools. (`pyproject` is to be preferred in cases where both are possible.)

Reply:
just out of curocity, what's the benefits to build with an explicit format.  couldn't find more information on format in nixpkgs manual.

Reply:
So `pyproject.toml` is preferred as it's more static and Python packaging is moving towards it where possible.  I suppose the `nixpkgs` maintainers prefer this to be made explicit to aid in moving towards pyproject-based builds where possible?

Reply:
I tried to build fastai with format=pyproject, and it built, but it doesn't have pyproject.toml in repo. so I guess it must not read from the pyproject.toml, otherwise the build will break. I set all these packages to setuptools, as they don't have pyproject.toml. but even if a project has a pyproject.toml, the version is still dictated by nixpkgs not pyproject.toml. right?

Reply:
Correct.